### ISSUE TITLE
refactor: Replace manual NLL with F.cross_entropy

### ISSUE THREAD

**Issue by muuki2** [muuki2/makemore-clone#9]
Use torch.nn.functional.cross_entropy instead of manual NLL loss to simplify the training pipeline.
